pkg/interface/chat/src/js/components/chat.js

```diff
@@ -46,77 +41,57 @@ export class ChatScreen extends Component {
   componentDidUpdate(prevProps, prevState) {
     const { props, state } = this;
 
-    if (prevProps.match.params.ship !== props.match.params.ship ||
-              prevProps.match.params.station !== props.match.params.station
-    ) {
-      console.log('switched circle');
+    if ((prevProps.match.params.station !== props.match.params.station) ||
+      (prevProps.match.params.ship !== props.match.params.ship)) {
       this.hasAskedForMessages = false;
 
       clearInterval(this.updateReadInterval);
 
       this.setState({
-        station: props.match.params.ship + "/" + props.match.params.station,
-        circle: props.match.params.station,
-        host: props.match.params.ship,
-        numPeople: 0,
+      station: `/${props.match.params.ship}/${props.match.params.station}`,
         scrollLocked: false
       }, () => {
-        this.updateNumPeople();
         this.scrollToBottom();
         this.updateReadInterval = setInterval(
           this.updateReadNumber.bind(this),
           1000
         );
         this.updateReadNumber();
       });
-    } else if (!(state.station in props.configs)) {
+    } else if (Object.keys(props.inbox) > 0) {
```

Comment:
It looks like this will always be `false`.

Reply:
Are you sure?  It looks like `props.inbox` should always be an object. `Object.keys` returns an array, which you're comparing against `0` (hence false).  Should there be a `.length` here, or am I missing something?

```
> let foo = { one: 1, two : 2 }
> let bar = {}
> Object.keys(foo) > 0
false
> Object.keys(bar) > 0
false
```

Reply:
Ah, I should add a `.length`. Good catch!

Reply:
Not only that, it needed to be changed to .length === 0 rather than > 0. Caught this while testing. Thanks!